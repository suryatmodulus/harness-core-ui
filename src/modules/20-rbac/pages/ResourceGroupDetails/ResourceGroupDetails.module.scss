/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.resourceTypeListContainer {
  background-color: var(--white) !important;
  box-shadow: var(--card-shadow);
}
.pageContainer {
  display: grid;
  grid-auto-flow: column;
  grid-template-columns: 320px 1fr;
}
.colorDiv {
  min-height: 40px;
  min-width: 40px;
  display: inline-block;
  border-radius: var(--spacing-xsmall);
}
