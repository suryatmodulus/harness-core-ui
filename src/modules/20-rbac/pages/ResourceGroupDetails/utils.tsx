import produce from 'immer'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector } from '@rbac/utils/utils'
import type { ResourceSelector } from 'services/resourcegroups'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (isDynamicResourceSelector(resourceSelector.type)) {
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    } else {
      map.set(resourceSelector.resourceType, resourceSelector.identifiers)
    }
  })
  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}
