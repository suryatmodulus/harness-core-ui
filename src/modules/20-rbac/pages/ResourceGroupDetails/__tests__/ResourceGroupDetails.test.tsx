/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, act, fireEvent, RenderResult, queryByAttribute, getByText } from '@testing-library/react'
import { findDialogContainer, TestWrapper } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import { accountPathProps, resourceGroupPathProps } from '@common/utils/routeUtils'
import {
  getResourceTypeHandlerMock,
  getResourceGroupTypeHandlerMock,
  getResourceCategoryListMock
} from '@rbac/utils/RbacFactoryMockData'
import ResourceGroupDetails from '../ResourceGroupDetails'
import { resourceTypes, resourceGroupDetails, resourceGroupDetailsWithHarnessManaged } from './mock'

const updateResourceGroupDetails = jest.fn()
const getResourceGroupDetailsMock = jest.fn().mockImplementation(() => {
  return { data: resourceGroupDetails, refetch: jest.fn(), error: null, loading: false }
})
const updateResourceGroupDetailsMock = (data: any): Promise<{ status: string }> => {
  updateResourceGroupDetails(data.resourcegroup)
  return Promise.resolve({ status: 'SUCCESS' })
}
jest.mock('services/resourcegroups', () => ({
  useGetResourceTypes: jest.fn().mockImplementation(() => {
    return { data: resourceTypes, refetch: jest.fn(), error: null, loading: false }
  }),
  useUpdateResourceGroup: jest.fn().mockImplementation(() => ({ mutate: updateResourceGroupDetailsMock })),
  useGetResourceGroup: jest.fn().mockImplementation(() => {
    return getResourceGroupDetailsMock()
  })
}))

jest.mock('@rbac/factories/RbacFactory', () => ({
  getResourceTypeHandler: jest.fn().mockImplementation(resource => getResourceTypeHandlerMock(resource)),
  getResourceCategoryHandler: jest.fn().mockImplementation(resource => getResourceGroupTypeHandlerMock(resource)),
  getResourceCategoryList: jest.fn().mockImplementation(() => getResourceCategoryListMock()),
  getResourceTypeLabelKey: jest.fn()
}))

describe('Resource Groups Page', () => {
  let renderObj: RenderResult
  beforeEach(() => {
    renderObj = render(
      <TestWrapper
        path={routes.toResourceGroupDetails({ ...accountPathProps, ...resourceGroupPathProps })}
        pathParams={{ accountId: 'dummy', resourceGroupIdentifier: 'dummyResourceGroupIdentifier' }}
      >
        <ResourceGroupDetails />
      </TestWrapper>
    )
  })
  afterEach(() => {
    renderObj.unmount()
  })
  test('render data', async () => {
    const { container } = renderObj
    expect(container).toMatchSnapshot()
  })
  test('test projects selection and save', async () => {
    const { getAllByText, container } = renderObj
    const project = queryByAttribute('data-testid', container, 'CHECK-BOX-PROJECT')
    expect(project).toBeTruthy()
    act(() => {
      fireEvent.click(project!)
    })
    expect(getAllByText('rbac.resourceGroup.all')[0]).toBeTruthy()
    await act(async () => {
      fireEvent.click(getByText(container, 'applyChanges'))
    })
    expect(updateResourceGroupDetails).toBeCalledWith({
      accountIdentifier: 'kmpySmUISimoRrJL6NL73w',
      orgIdentifier: null,
      projectIdentifier: null,
      identifier: 'ewrewew',
      name: 'nameewrewew',
      resourceSelectors: [
        { type: 'DynamicResourceSelector', resourceType: 'ORGANIZATION' },
        { type: 'DynamicResourceSelector', resourceType: 'PROJECT' }
      ],
      tags: {},
      description: '',
      color: '#0063f7'
    })
  })
  test('test orgs selection and save', async () => {
    const { container, getByTestId } = renderObj
    const specifiedResource = getByTestId('static-ORGANIZATION')
    act(() => {
      fireEvent.click(specifiedResource)
    })
    const addResources = getByTestId('addResources-ORGANIZATION')
    expect(addResources).toBeTruthy()
    act(() => {
      fireEvent.click(addResources)
    })
    let form = findDialogContainer()
    expect(form).toBeTruthy()

    act(() => {
      fireEvent.click(getByText(form!, 'cancel'))
    })
    form = findDialogContainer()
    expect(form).toBeFalsy()

    await act(async () => {
      fireEvent.click(getByText(container, 'applyChanges'))
    })
    expect(updateResourceGroupDetails).toBeCalledWith({
      accountIdentifier: 'kmpySmUISimoRrJL6NL73w',
      orgIdentifier: null,
      projectIdentifier: null,
      identifier: 'ewrewew',
      name: 'nameewrewew',
      resourceSelectors: [
        { type: 'DynamicResourceSelector', resourceType: 'ORGANIZATION' },
        { type: 'DynamicResourceSelector', resourceType: 'PROJECT' }
      ],
      tags: {},
      description: '',
      color: '#0063f7'
    })
  })
})
test('with harness managed resources', async () => {
  getResourceGroupDetailsMock.mockImplementation(() => {
    return { data: resourceGroupDetailsWithHarnessManaged, refetch: jest.fn(), error: null, loading: false }
  })
  const { queryByText } = render(
    <TestWrapper
      path={routes.toResourceGroupDetails({ ...accountPathProps, ...resourceGroupPathProps })}
      pathParams={{ accountId: 'dummy', resourceGroupIdentifier: 'dummyResourceGroupIdentifier' }}
    >
      <ResourceGroupDetails />
    </TestWrapper>
  )
  const applyChanges = queryByText('applyChanges')
  expect(applyChanges).toBeFalsy()
})
