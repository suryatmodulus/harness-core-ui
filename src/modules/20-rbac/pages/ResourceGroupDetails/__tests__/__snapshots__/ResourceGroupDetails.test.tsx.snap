// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Groups Page render data 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--xlarge StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="PageHeader--breadcrumbsDiv"
      >
        <div
          class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <a
              class="Breadcrumbs--breadcrumb"
              href="/account/dummy/settings"
            >
              <span
                class="bp3-icon bp3-icon-cog StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary7"
                icon="cog"
              >
                <svg
                  data-icon="cog"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    cog
                  </desc>
                  <path
                    d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
              >
                common.accountSettings
              </p>
            </a>
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
              data-icon="main-chevron-right"
            >
              <svg
                height="8"
                viewBox="0 0 8 14"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <a
              class="Breadcrumbs--breadcrumb"
              href="/account/dummy/settings/access-control/resource-groups"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
              >
                accessControl: resourceGroups
              </p>
            </a>
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
              data-icon="main-chevron-right"
            >
              <svg
                height="8"
                viewBox="0 0 8 14"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start StyledProps--flex-alignItems-center"
      >
        <div
          class="colorDiv"
          style="background-color: rgb(0, 99, 247);"
        />
        <div
          class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-left-small"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-medium StyledProps--color StyledProps--color-black"
            style="--text-line-clamp: 1; width: 700px;"
          >
            nameewrewew
          </p>
          
        </div>
      </div>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--margin-top-huge"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-right-small StyledProps--border StyledProps--border-right StyledProps--border-color-grey300"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-small"
        >
          CREATED
          :
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-small"
          style="--text-line-clamp: 1;"
        >
          dummy date
        </p>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-left-small"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-small"
        >
          LASTUPDATED
          :
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-small"
          style="--text-line-clamp: 1;"
        >
          dummy date
        </p>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody resourceGroupDetails"
    data-name="page-body"
  >
    <div
      class="bp3-card bp3-elevation-0 Card--card subHeader"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between"
        style="width: 100%;"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
        >
          rbac.resourceGroup.selectionHeading
        </p>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-end"
        >
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              applyChanges
            </span>
          </button>
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              common.discard
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main pageContainer"
    >
      <div
        class="StyledProps--font StyledProps--main resourceTypeListContainer StyledProps--padding-xlarge"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-small StyledProps--main resourceTypeList StyledProps--flex"
        >
          <div
            class="Layout--horizontal StyledProps--main resourcePicker StyledProps--flex StyledProps--padding-bottom-small"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-h6 StyledProps--color StyledProps--color-grey800"
            >
              resources
            </p>
            <div
              class="Layout--horizontal Layout--layout-spacing-huge StyledProps--main StyledProps--flex"
            >
              <label
                class="bp3-control bp3-radio StyledProps--font Radio--radio StyledProps--main StyledProps--inline"
              >
                <input
                  type="radio"
                  value="ALL"
                />
                <span
                  class="bp3-control-indicator"
                />
                rbac.resourceGroup.all
              </label>
              <label
                class="bp3-control bp3-radio StyledProps--font Radio--radio StyledProps--main StyledProps--inline"
              >
                <input
                  checked=""
                  type="radio"
                  value="SPECIFIED"
                />
                <span
                  class="bp3-control-indicator"
                />
                common.specified
              </label>
            </div>
          </div>
          <div
            class="bp3-card bp3-elevation-0 Card--card resourceTypeCard"
            draggable="true"
          >
            <div
              class="Layout--vertical StyledProps--main"
            >
              <div
                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkBox StyledProps--main"
                >
                  <input
                    data-testid="CHECK-BOX-ADMINSTRATIVE_FUNCTIONS"
                    type="checkbox"
                    value="ADMINSTRATIVE_FUNCTIONS"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
                  style="display: flex; align-items: center;"
                >
                  <span
                    class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-small"
                    data-icon="nav-project"
                  >
                    <svg
                      height="16"
                      viewBox="0 0 22 22"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.1 4.583L11 0l9.9 4.583v12.834L11 22l-9.9-4.583zm9 5.171l-7.2-3.33v9.811l7.2 3.338V9.754zm9 6.481V6.424l-7.2 3.33v9.819zm-1.462-11.15L11 2.012 4.362 5.085 11 8.157z"
                        fill="#56778d"
                      />
                    </svg>
                  </span>
                  ADMINFUNCTIONS
                </p>
              </div>
              <div
                class="Layout--horizontal StyledProps--main resourceSubList StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    data-testid="CHECK-BOX-ORGANIZATION"
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-small StyledProps--color StyledProps--color-grey800"
                >
                  orgsText
                </p>
              </div>
              <div
                class="Layout--horizontal StyledProps--main resourceSubList StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    data-testid="CHECK-BOX-PROJECT"
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-small StyledProps--color StyledProps--color-grey800"
                >
                  projectsText
                </p>
              </div>
            </div>
          </div>
          <div
            class="bp3-card bp3-elevation-0 Card--card resourceTypeCard"
            draggable="true"
          >
            <div
              class="Layout--vertical StyledProps--main"
            >
              <div
                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox checkBox StyledProps--main"
                >
                  <input
                    data-testid="CHECK-BOX-SHARED_RESOURCES"
                    type="checkbox"
                    value="SHARED_RESOURCES"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600"
                  style="display: flex; align-items: center;"
                >
                  <span
                    class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-small"
                    data-icon="nav-project"
                  >
                    <svg
                      height="16"
                      viewBox="0 0 22 22"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.1 4.583L11 0l9.9 4.583v12.834L11 22l-9.9-4.583zm9 5.171l-7.2-3.33v9.811l7.2 3.338V9.754zm9 6.481V6.424l-7.2 3.33v9.819zm-1.462-11.15L11 2.012 4.362 5.085 11 8.157z"
                        fill="#56778d"
                      />
                    </svg>
                  </span>
                  RBAC.CATEGORYLABELS.SHAREDRESOURCES
                </p>
              </div>
              <div
                class="Layout--horizontal StyledProps--main resourceSubList StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    checked=""
                    data-testid="CHECK-BOX-SECRET"
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-small StyledProps--color StyledProps--color-grey800"
                >
                  common.secrets
                </p>
              </div>
              <div
                class="Layout--horizontal StyledProps--main resourceSubList StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    data-testid="CHECK-BOX-CONNECTOR"
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-small StyledProps--color StyledProps--color-grey800"
                >
                  connectorsLabel
                </p>
              </div>
              <div
                class="Layout--horizontal StyledProps--main resourceSubList StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
              >
                <label
                  class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
                >
                  <input
                    data-testid="CHECK-BOX-DELEGATE"
                    type="checkbox"
                  />
                  <span
                    class="bp3-control-indicator"
                  />
                </label>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-small StyledProps--color StyledProps--color-grey800"
                >
                  delegate.delegates
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-xlarge"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start StyledProps--padding-bottom-small"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-h6 StyledProps--color StyledProps--color-grey800"
            >
              common.scope
            </p>
            <div
              class="bp3-popover-wrapper DropDown--main"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class=""
                >
                  <div
                    class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                    data-testid="dropdown-button"
                    style="width: 300px;"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                    >
                      <span
                        class="bp3-icon bp3-icon-settings StyledProps--main StyledProps--color StyledProps--color-grey600"
                        icon="settings"
                      >
                        <svg
                          data-icon="settings"
                          height="13"
                          viewBox="0 0 16 16"
                          width="13"
                        >
                          <desc>
                            settings
                          </desc>
                          <path
                            d="M3 1c0-.55-.45-1-1-1S1 .45 1 1v3h2V1zm0 4H1c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm12-4c0-.55-.45-1-1-1s-1 .45-1 1v2h2V1zM9 1c0-.55-.45-1-1-1S7 .45 7 1v6h2V1zM1 15c0 .55.45 1 1 1s1-.45 1-1v-5H1v5zM15 4h-2c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V5c0-.55-.45-1-1-1zm-2 11c0 .55.45 1 1 1s1-.45 1-1V9h-2v6zM9 8H7c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h2c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-2 7c0 .55.45 1 1 1s1-.45 1-1v-2H7v2z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                        data-testid="dropdown-value"
                        style="--text-line-clamp: 1;"
                      >
                        rbac.scopeItems.accountOnly
                      </p>
                    </div>
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                    >
                      <span
                        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                        data-icon="main-chevron-down"
                      >
                        <svg
                          height="8"
                          viewBox="0 0 14 8"
                          width="8"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                            fill="#4d5969"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="bp3-card bp3-elevation-0 Card--card selectedResourceGroupCardDetails"
          >
            <div
              class="Layout--vertical StyledProps--main"
            >
              <div
                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-center"
              >
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--font-weight-semi-bold"
                    style="display: flex; align-items: center;"
                  >
                    <span
                      class="bp3-icon bp3-icon-lock StyledProps--main Text--icon StyledProps--padding-right-medium"
                      icon="lock"
                    >
                      <svg
                        data-icon="lock"
                        height="30"
                        viewBox="0 0 20 20"
                        width="30"
                      >
                        <desc>
                          lock
                        </desc>
                        <path
                          d="M15.93 9H14V4.99c0-2.21-1.79-4-4-4s-4 1.79-4 4V9H3.93c-.55 0-.93.44-.93.99v8c0 .55.38 1.01.93 1.01h12c.55 0 1.07-.46 1.07-1.01v-8c0-.55-.52-.99-1.07-.99zM8 9V4.99c0-1.1.9-2 2-2s2 .9 2 2V9H8z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                    common.secrets
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main StyledProps--flex"
                >
                  <label
                    class="bp3-control bp3-radio StyledProps--font Radio--radio StyledProps--main"
                  >
                    <input
                      checked=""
                      data-testid="dynamic-SECRET"
                      type="radio"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    rbac.resourceGroup.all
                  </label>
                  <div
                    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main radioBtn StyledProps--flex StyledProps--padding-left-huge"
                  >
                    <label
                      class="bp3-control bp3-radio StyledProps--font Radio--radio StyledProps--main"
                    >
                      <input
                        data-testid="static-SECRET"
                        type="radio"
                      />
                      <span
                        class="bp3-control-indicator"
                      />
                      common.specified
                    </label>
                    <button
                      class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main addResourceBtn Button--with-current-color Button--iconOnly Button--variation Button--variation-link"
                      data-testid="addResources-SECRET"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="bp3-button-text"
                      >
                        rbac.resourceGroup.add
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
